components/datetime/src/raw/datetime.rs

```diff
@@ -27,10 +31,299 @@ use icu_locid::Locale;
 use icu_plurals::{provider::OrdinalV1Marker, PluralRules};
 use icu_provider::prelude::*;
 
-use crate::{
-    date::DateTimeInput, pattern::runtime::PatternPlurals, provider, DateTimeFormatError,
-    FormattedDateTime,
-};
+pub(crate) struct TimeFormat {
+    pub locale: Locale,
+    pub patterns: DataPayload<PatternPluralsFromPatternsV1Marker>,
+    pub fixed_decimal_format: FixedDecimalFormat,
+}
+
+impl TimeFormat {
+    /// Constructor that takes a selected [`Locale`], reference to a [`DataProvider`] and
+    /// a list of options, then collects all data necessary to format date and time values into the given locale,
+    /// using the short style.
+    ///
+    /// The "calendar" argument should be a Unicode BCP47 calendar identifier
+    #[inline(never)]
+    pub fn try_new<D>(
+        locale: Locale,
+        data_provider: &D,
+        length: length::Time,
+        preferences: Option<preferences::Bag>,
+    ) -> Result<Self, DateTimeFormatError>
+    where
+        D: ResourceProvider<TimePatternsV1Marker>
+            + ResourceProvider<DecimalSymbolsV1Marker>
```

Comment:
Observation: Time symbols should be its own key

Reply:
Thanks for catching this!